components/script/dom/range.rs

```diff
@@ -658,6 +658,9 @@ impl RangeMethods for Range {
         let start_offset = self.StartOffset();
 
         // Step 1.
+        if &*start_node == node {
+            return Err(Error::HierarchyRequest);
+        }
         match start_node.type_id() {
             // Handled under step 2.
             NodeTypeId::CharacterData(CharacterDataTypeId::Text) => (),
```

Comment:
This is already covered in step 2, as mentioned in the comment above the line you commented. :)